src/main/java/com/github/manolo8/darkbot/core/utils/ByteUtils.java

```diff
@@ -222,5 +226,42 @@ public void resetCache() {
         public int getVersion() {
             return 1;
         }
+
+        private byte[] tableData = new byte[0];
+        private long lastTableUpdate;
+        /**
+         * @author Alph4rd
+         */
+        @Override
+        public long searchClassClosure(Predicate<Long> pattern) {
+            long mainAddress = botInstaller.mainApplicationAddress.get();
+            if (mainAddress == 0) return 0;
+
+            long table = Main.API.readMemoryLong(mainAddress, 0x10, 0x10, 0x18, 0x10, 0x28);
+            int capacity = Main.API.readMemoryInt(table + 8) * 8;
+
+            if (tableData.length < capacity) {
+                tableData = new byte[capacity];
```

Comment:
We're allocating a table of arbitrary size, this is generally a pretty bad idea

The mem-read could return a random value that we cannot fullfil, causing an out of mem exception directly
Also, this table is not cleared after 750ms, it is only *replaced* after 750ms if anything happens to re-request it, but if nothing asks for this table again it will just stay there forever taking up however much memory it required